salvo/src/lib/benchmark/fully_dockerized_benchmark.py

```diff
@@ -0,0 +1,37 @@
+"""
+This module contains the methods to perform a nighthawk benchmark using
+containers for the scripts, nighthawk binaries, and envoy
+
+https://github.com/envoyproxy/nighthawk/blob/master/benchmarks/README.md
+"""
+
+import logging
+
+from src.lib.benchmark.base_benchmark import BaseBenchmark
+
+log = logging.getLogger(__name__)
+
+
+class Benchmark(BaseBenchmark):
+
+  def __init__(self, **kwargs):
+    super(Benchmark, self).__init__(**kwargs)
+
+  def validate(self):
```

Comment:
Since this is a public method, are we expecting the callers to call it specifically? Is that the right approach or should it be a private method that we always call from say `execute_benchmark`?

Other way of asking the same is - are there any situations where the caller would want to execute without validating? If we call it internally we may get better safety (impossible to execute without validation) and smaller API surface.

Reply:
Right, this is true.  We shouldn't rely on the user of the class to call validate.  I'll make this private and update the PR